### ISSUE TITLE
non-transparent selection covers text

### ISSUE THREAD

**Issue by UziTech** [xtermjs/xterm.js#2737]
Setting selection color to non-transparent color (e.g. `#0000ff`) covers text in canvas but not when using webgl addon.

## Details
- Browser and browser version: Chrome 80.0.3987.122
- OS version: Windows 10
- xterm.js version: 4.4.0

### Steps to reproduce

1. Go To https://codepen.io/UziTech/pen/gOpmPGo
2. Select text in each terminal

![image](https://user-images.githubusercontent.com/97994/75286256-d455c080-57dd-11ea-95c3-ebedab87ed83.png)

**Comment by Tyriar** [comment]
This is kind of by design. For the normal renderers this was always a limitation as there was no efficient way of doing it, monaco has a similar issue where text background is not supported at all for performance reasons. So for historic reasons the selection color is meant to be transparent.

The WebGL renderer however was finally able to do it efficiently so we can draw the selection behind the text (and even change selection text color for a11y reasons if we want) so opaque colors work best.

The problem is though then there's an inconsistency in how the color is presented. My thinking to solve this was to blend the color with the background manually (0.3 opacity or something?) when an opaque color is provided such that the text is still visible. This is a problem in VS Code that I haven't gotten around to solving as well; using webgl and switching off it often necessitates switching my theme override color which currently looks like this:

```
  "workbench.colorCustomizations": {
    "[Default Light+]": {
      "terminal.selectionBackground": "#add6ff"
    },
    "[Visual Studio Light]": {
      "terminal.selectionBackground": "#add6ff"
    },
    "[Sapphire]": {
      "terminal.selectionBackground": "#153958"
    },
    "[Topaz (Dim)]": {
      "terminal.selectionBackground": "#42400f"
    }
  }
```

TL;DR: Let's change the canvas and dom renderers to blend the color with the background color at 0.3 opacity if the selection color is opaque.

**Comment by Puneethnaik** [comment]
@Tyriar I would like to take this issue. Upon first inspection, I figured we need to make changes in the setTheme function of ColorManager.ts
https://github.com/xtermjs/xterm.js/blob/869d791f444f77024d97af0e2fe430996259a937/src/browser/ColorManager.ts#L113
And also, I am thinking there should be some changes in the color.blend function as well. 
https://github.com/xtermjs/xterm.js/blob/869d791f444f77024d97af0e2fe430996259a937/src/browser/Color.ts#L29
Please guide me further.

**Comment by Tyriar** [comment]
@Puneethnaik also sorry about not giving any guidance, been 100+ notifications behind for a while 😩

**Comment by Puneethnaik** [comment]
@Tyriar No worries :slightly_smiling_face:, I have got your reviews on the PR. It has taught me many things about good software engineering practices.
